event-based-simulators/main/event_based_simulators.py

```diff
@@ -419,20 +384,88 @@ def __send_event(self, event_fragment, timestamp = None):
             return newTimestamp
 
     def is_in_productionTime(self):
-        #if there are no shiftplans for a device, it should not be affected by production-time
+        # if there are no shiftplans for a device, it should not be affected by production-time
         has_no_shiftplan = True
         for shiftplan in shiftplans:
             if shiftplan.locationId == self.locationId:
                 has_no_shiftplan = False
                 for timeslot in shiftplan.recurringTimeSlots:
                     if timeslot.slotType == "PRODUCTION":
                         now = datetime.utcnow()
-                        start = datetime.strptime(timeslot.slotStart, shiftplan_dateformat)
-                        end = datetime.strptime(timeslot.slotEnd, shiftplan_dateformat)
+                        start = datetime.strptime(timeslot.slotStart, Shiftplan.dateformat)
+                        end = datetime.strptime(timeslot.slotEnd, Shiftplan.dateformat)
                         if start < now and end > now:
                             return True
         return has_no_shiftplan
 
+    # Measurements functions #
+    def measurement_functions(self, measurement_definition, task):
+        MachineSimulator.generate_measurement(self=self, measurement_definition=measurement_definition)
+        MachineSimulator.send_measurements(self=self, measurement_definition=measurement_definition)
+
+    def generate_measurement(self, measurement_definition):
+        log.info(f"Generating value of measurement {measurement_definition.get('series')} of device {self.model.get('id')}")
+        self.simulated_data = []
+        distribution = measurement_definition.get("valueDistribution", "uniform")
+        value = 0.0
+        if distribution == "uniform":
+            min_value = measurement_definition.get("minimumValue", measurement_definition.get("value"))
+            max_value = measurement_definition.get("maximumValue", measurement_definition.get("value"))
+            value = round(uniform(min_value, max_value), 2)
+        elif distribution == "uniformint":
+            min_value = measurement_definition.get("minimumValue", measurement_definition.get("value"))
+            max_value = measurement_definition.get("maximumValue", measurement_definition.get("value"))
+            value = randint(min_value, max_value)
+        elif distribution == "normal":
+            mu = measurement_definition.get("mu")
+            sigma = measurement_definition.get("sigma")
+            value = round(gauss(mu, sigma), 2)
+        self.simulated_data.append({
+            'type': measurement_definition.get("type"),
+            'series': measurement_definition.get("series"),
+            'value': value,
+            'unit': measurement_definition.get("unit"),
+            'time': self.current_time,
+        })
+
+    def send_measurements(self, measurement_definition):
+        json_measurements_list = []
```

Comment:
why do we need to create an array if we only ever add a single element?

Reply:
I created this method when thinking about sending a collection of measurements but then I changed to send only a single one so it is still left like that. Since it doesn't affect the code until now so I forget about it